### PR TITLE
missiles collide and legend appears with key hints

### PR DESCRIPTION
**Pull request by sashithedog**
the missiles now collide properly and the legend pops up at the bottom of the screen and goes away when you press a key

### CODE REVIEW

File: missile.pde [hunk 1]
```diff
@@ -23,15 +23,15 @@ class Missile {
...
+  int lifeSpan = 500;
   float rot; // where the missile is facing
   Ship parent;
   Ship enemyShip;
   float enemyRange = windowSize; // can track you over entire screen right now
   float enemyAngleTolerance = 360;
   float mass = .4;
   float missileLaunchForce = 1;
-  float missileSmartFactor = 0.046;
+  float missileSmartFactor = 0.045;
```
Comment (willkessler) [inline missile.pde:34]: why did you make the missile smarter, dude?

Comment (sashithedog) [reply missile.pde:34]: i made it less smart

File: ship.pde [hunk 2]
```diff
@@ -198,6 +198,14 @@ class Ship {
     boolean impact = (((abs(pos.x - missilePos.x) < 10) && (abs(pos.y - missilePos.y) < 10)));
     return impact && otherShip.missile.isLive();
   }
+  
+  
+   boolean missileHitOtherShipsMissile(Ship otherShip) {
```
Comment (willkessler) [inline ship.pde:203]: nicely done! love this new method here.

File: spacewar.pde [hunk 3]
```diff
@@ -200,10 +201,12 @@ println("this should have worked out ");
 void draw()
 {
   background(0,0,0);
-  theStars.render();
+    theStars.render();
```
Comment (willkessler) [inline spacewar.pde]: get rid of this extra space indent

**Review by willkessler — APPROVED** [review]
LGTM!!

**Pull request merged**